docs/development.md

```diff
@@ -0,0 +1,66 @@
+# Development
+
+This action's core functionality is contained in a Go CLI. The `action.yml` at the root of
+this repo defines the action itself, as a composite action, that builds and then calls the CLI.
+
+## Manually Testing the CLI
+
+It's useful to be able to just run the CLI locally to manually poke around and see your changes.
+
+The workflow for this is to run `make install` any time you want to test your local changes,
+this runs tests and does a fully dogfooded from-scratch build and installs it into your PATH.
```

Comment:
I think it's worth mentioning here that the destination can be overridden with `make install DESTDIR=...`.